apps/meteor/app/api/server/v1/channels.ts

```diff
@@ -565,22 +565,36 @@ API.v1.addRoute(
 	{ authRequired: true },
 	{
 		async post() {
-			const findResult = await findChannelByIdOrName({
-				params: this.bodyParams,
-				checkedArchived: false,
-			});
+			let roomId = 'roomId' in this.bodyParams ? this.bodyParams.roomId : undefined;
+			let roomName = 'roomName' in this.bodyParams ? this.bodyParams.roomName : undefined;
 
-			const sub = await Subscriptions.findOneByRoomIdAndUserId(findResult._id, this.userId);
+			try {
+				const findResult = await findChannelByIdOrName({
+					params: this.bodyParams,
+					checkedArchived: false,
+				});
+				roomName = findResult.name;
+
+				if (findResult._id) {
+					roomId = findResult._id;
+				}
```

Comment:
```suggestion
				const roomName = this.bodyParams.roomName || findResult.name;
				const roomId = this.bodyParams.roomName || findResult._id;
```
We usually avoid using `let`. Also, you don't need to check if `_id` is defined since this is checked inside `findChannelByIdOrName`, which returns an `IRoom` object.
By applying these changes, you can delete the `let` definitions above.

Reply:
Here as the code is inside the `try` block, it wouldn't be possible to access the `roomName` and `roomId` from the outside, would it? About the `_id`, that is because if the room collection was dropped, the IRoom would not be found 🤔

Reply:
Since roomId and roomName are not used in the try/catch blockm you can still avoid using let by defining roomId and roomName after the search. So you could do something like:
`const roomId = findResult._id || this.bodyParams.roomId;` (after and outside of the try/catch)